esrally/driver/runner.py

```diff
@@ -1320,46 +1329,56 @@ async def _wait_for(self, es, idx, description):
-                                              "index.blocks.write": "true"
-                                          }
-                                      },
-                                      preserve_existing=True)
-
-        self.logger.info("Waiting for relocation to finish for index [%s]...", source_index)
-        await self._wait_for(es, source_index, "shard relocation for index [{}]".format(source_index))
-        self.logger.info("Shrinking [%s] to [%s].", source_index, target_index)
-        if "settings" not in target_body:
-            target_body["settings"] = {}
-        target_body["settings"]["index.routing.allocation.require._name"] = None
-        target_body["settings"]["index.blocks.write"] = None
-        # kick off the shrink operation
-        await es.indices.shrink(index=source_index, target=target_index, body=target_body)
-
-        self.logger.info("Waiting for shrink to finish for index [%s]...", source_index)
-        await self._wait_for(es, target_index, "shrink for index [{}]".format(target_index))
-        self.logger.info("Shrinking [%s] to [%s] has finished.", source_index, target_index)
+            self.logger.info("Using [%s] as shrink node.", shrink_node)
+            self.logger.info("Preparing [%s] for shrinking.", source_index)
+
+            # prepare index for shrinking
+            await es.indices.put_settings(index=source_index,
+                                          body={
+                                              "settings": {
+                                                  "index.routing.allocation.require._name": shrink_node,
+                                                  "index.blocks.write": "true"
+                                              }
+                                          },
+                                          preserve_existing=True)
+
+            self.logger.info("Waiting for relocation to finish for index [%s] ...", source_index)
+            await self._wait_for(es, source_index, "shard relocation for index [{}]".format(source_index))
+            self.logger.info("Shrinking [%s] to [%s].", source_index, target_index)
+            if "settings" not in target_body:
+                target_body["settings"] = {}
+            target_body["settings"]["index.routing.allocation.require._name"] = None
+            target_body["settings"]["index.blocks.write"] = None
+            # kick off the shrink operation
+            index_suffix = remove_prefix(source_index, source_indices_stem)
+            final_target_index = target_index if len(index_suffix) == 0 else target_index+index_suffix
+            await es.indices.shrink(index=source_index, target=final_target_index, body=target_body)
+
+            self.logger.info("Waiting for shrink to finish for index [%s] ...", source_index)
+            await self._wait_for(es, final_target_index, "shrink for index [{}]".format(final_target_index))
```

Comment:
We could run things in parallel by moving this to a separate loop later, however, I am not sure whether a high level of concurrent shrink operations will actually be beneficial or cause contention and provide minimal advantage (esp. if only one shrink node has been provided).

Reply:
Agreed, I'd not complicate things unless there is a strong benefit.